libs/s25main/ai/random.cpp

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2005 - 2025 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "ai/random.h"
+
+namespace AI {
+
+std::minstd_rand& getRandomGenerator()
+{
+    std::random_device rnd_dev;
+    static std::minstd_rand rng(rnd_dev());
```

Comment:
This would construct a random_device even when not required. Not sure if this does anything but better be sure:
```suggestion
    static std::minstd_rand rng(std::random_device{}());
```